include/revng/Model/CMakeLists.txt

```diff
@@ -0,0 +1,89 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+# Header files containing model YAML must be added here
+set(MODEL_HEADERS
+    CABIFunctionType.h
+    EnumType.h
+    PrimitiveType.h
+    PrimitiveTypeKind.h
+    RawFunctionType.h
+    StructType.h
+    TypedefType.h
+    Type.h
+    TypeKind.h
+    UnionType.h
+    ABI.h
+    AbstractFunctionEdge.h
+    Architecture.h
+    Argument.h
+    BasicBlock.h
+    Binary.h
+    CallEdge.h
+    DynamicFunction.h
+    EnumEntry.h
+    FunctionEdge.h
+    FunctionEdgeType.h
+    Function.h
+    FunctionType.h
+    NamedTypedRegister.h
+    QualifiedType.h
+    Qualifier.h
+    QualifierKind.h
+    Register.h
+    RegisterState.h
+    Segment.h
+    StructField.h
+    TypedRegister.h
+    UnionField.h
+)
+
+set(MODEL_TEMPLATES
+    "${CMAKE_SOURCE_DIR}/scripts/tuple_tree_generator/tuple_tree_generator/templates/class_forward_decls.tpl.hpp"
+    "${CMAKE_SOURCE_DIR}/scripts/tuple_tree_generator/tuple_tree_generator/templates/enum.tpl.hpp"
+    "${CMAKE_SOURCE_DIR}/scripts/tuple_tree_generator/tuple_tree_generator/templates/struct.tpl.hpp"
+    "${CMAKE_SOURCE_DIR}/scripts/tuple_tree_generator/tuple_tree_generator/templates/struct_forward_decls.tpl.hpp"
+    "${CMAKE_SOURCE_DIR}/scripts/tuple_tree_generator/tuple_tree_generator/templates/struct_late.tpl.hpp"
+    "${CMAKE_SOURCE_DIR}/scripts/tuple_tree_generator/tuple_tree_generator/templates/struct_impl.tpl.cpp"
+)
+
+# -- Collect YAML from manually written headers
+set(COLLECTED_YAML_PATH "${CMAKE_BINARY_DIR}/include/revng/Model/model_schema.yml")
+
+add_custom_command(
+    OUTPUT "${COLLECTED_YAML_PATH}"
+    COMMAND "${CMAKE_SOURCE_DIR}/scripts/tuple_tree_generator/extract_yaml.py"
+    ARGS "MODEL-YAML" ${MODEL_HEADERS} > "${COLLECTED_YAML_PATH}"
+    WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
```

Comment:
Needed?

Reply:
Yes, first I need to collect all the YAML definitions from the headers, then those get processed to generate the C++ and the JSON schema.

Reply:
I think I meant `WORKING_DIRECTORY`. Let's avoid using the cwd unless necessary.

Reply:
Fair, removed